### PR TITLE
Add a modal to display remediation info on check results

### PR DESCRIPTION
**Pull request by rtorrero**
This PR adds a modals where users can check the remediation details from the check results screen by clicking on the check result row. In action:

remediation-info

### CODE REVIEW

**Review by abravosuse — APPROVED** [review]
Leaving code aside, this looks awesome @rtorrero !

File: assets/js/components/ClusterDetails/ChecksResults.jsx [hunk 1]
```diff
@@ -69,6 +73,14 @@ export const ChecksResults = () => {
     ]
   );
 
+  const checksRemediationById = Object.assign(
```
Comment (arbulu89) [inline assets/js/components/ClusterDetails/ChecksResults.jsx:76]: This approach works, but I find more "normal" to run a `find` function:
```
export getRemediationById = (id) => (catalogData) =>
  catalogData.flat().find((check) => check.id === id);
```
(didn't test, but I guess you get the idea)

File: assets/js/components/ClusterDetails/ChecksResults.jsx [hunk 2]
```diff
@@ -54,6 +56,8 @@ const getResultIcon = (executionState, health) => {
 };
 
 export const ChecksResults = () => {
+  const [modalOpen, setModalOpen] = useState(false);
+  const [selectedCheck, setSelectedCheck] = useState(false);
```
Comment (arbulu89) [inline assets/js/components/ClusterDetails/ChecksResults.jsx:60]: Maybe better to default to an empty string or null? This value is a string after all, and not a boolean
`useState("")`

**Review by arbulu89 — COMMENTED** [review]
The code looks good overall.
Not a big fan of the modals though, more when we are using collapsible rows everywhere else.

But if the PO like, I'm fine with it.
You have green light from me, just have a look on my comments, and feel free to change (if you want of course hehe)

PD: I guess that at some point we could use this information box to show some more information about why the check failed too

**Comment by abravosuse** [comment]
> PD: I guess that at some point we could use this information box to show some more information about why the check failed too

Me linking this idea VERY MUCH :-) I will add a user story for it...

**Pull request merged**